zio-http/src/main/scala/zhttp/http/Cookie.scala

```diff
@@ -125,6 +129,29 @@ final case class Cookie(
     cookie.flatten.mkString("; ")
   }
 
+  def sign(secret: String): Option[Cookie] = {
+    try {
+      val sha256    = Mac.getInstance("HmacSHA256")
+      val secretKey = new SecretKeySpec(secret.getBytes(), "RSA")
+      sha256.init(secretKey)
+      val signed    = sha256.doFinal(self.content.getBytes())
+      val mda       = MessageDigest.getInstance("SHA-512")
+      val content   = self.content + '.' + getEncoder.encodeToString(mda.digest(signed))
+      Some(self.withContent(content))
+    } catch {
+      case _: Exception => None
+    }
+
+  }
```

Comment:
Can we move this logic to the cookie encoder? Have a new field in `Cookie` called `secret: Option[String]`. That way if you have it in the cookie, you can use it to encode and sign the content.

Reply:
what about decodeResponseCookie? We won't be able to decode from content if it's signed or not, or what is the secret value. We can unsign using secret only.

Reply:
we can add another method to `decodeResponseSignedCookie` no?

Reply:
We shouldn't be so eager to sign the cookie, consider this example — 
```scala
val cookie = Cookie("A", "B").sign("secret").withContent("C")
```

The fact that this cookie should be signed is ignored when we set the content again. This IMO is counterintuitive and breaks compositional semantics.

What do you think?

Reply:
Yup, I feel the same. I've tried implementing it here: https://github.com/dream11/zio-http/pull/751/files